### PR TITLE
Feature/8 backend to create account

### LINKED ISSUE #8: Backend to create account
Asana task: https://app.asana.com/0/1204166279674278/1208203113907818/f

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] Enforce email validation
- [x] Enforce gender and reason for account creation options
- [x] Use system user Django table and extend it for new columns
- [x] FE Implementation
- [x] Send email to request people to confirm email before login will be possible
- [x] Use mailjet to send the email - integrate this
- [x] Update backend tests

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208203113907818

### CODE REVIEW

File: backend/api/v1/v1_users/models.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        default=None,
+        null=True
+    )
+    gender = models.IntegerField(
+        choices=Gender.FieldStr.items(),
+        default=None,
+        null=True
+    )
+    is_verified = models.BooleanField(default=False)
+    verification_token = models.TextField()
```
Comment (dedenbangkit) [inline backend/api/v1/v1_users/models.py]: UUID unique

File: backend/api/v1/v1_users/serializers.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    gender = CustomChoiceField(choices=Gender.FieldStr)
+    country = CustomCharField()
+    account_purpose = CustomChoiceField(choices=AccountPurpose.FieldStr)
+    email = CustomEmailField()
+    password = CustomCharField()
+    confirm_password = CustomCharField()
+    agreement = CustomBooleanField(default=False)
+
+    def validate(self, attrs):
+        if attrs.get("password") != attrs.get("confirm_password"):
```
Comment (dedenbangkit) [inline backend/api/v1/v1_users/serializers.py]: validation follow the FE rules

**Review by dedenbangkit — CHANGES_REQUESTED** [review]

**Review by dedenbangkit — CHANGES_REQUESTED** [review]
remove first_name and last_name, use full_name instead

**Comment by ifirmawan** [comment]
> remove first_name and last_name, use full_name instead

Resolved with the beginning of this commit [4e45256](https://github.com/akvo/partos-pat/pull/9/commits/4e45256885ed99b570a3f7f5db5f810785dd92f1)

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**